### PR TITLE
Save pronoun messages

### PR DESCRIPTION
**Pull request by daniel-panhead**
# TL;DR
Reactions to pronoun messages now work after restarting the bot

# Description
When the `/pronoun` command is run, it now saves a copy of the message ID and channel ID to Firebase under the `InitBotInfo > [server ID] > SavedMessages -> pronouns` document. When the bot is restarted, it will check for this document, and if there is an existing message/channel saved, it will use those to start emoji reaction listeners, allowing the old message to continue to be used for pronoun role reactions.

In addition, the bot startup was also changed so that each server is initialized sequentially instead of in parallel. This just makes the debug messages less confusing since they're not being logged asynchronously.

## Why
If the bot crashes, we will lose emoji reaction listeners on messages that have already been sent. This change will allow us to more easily recover from bot crashes.

## User Changes
None :D

## Tests Done
Tested restarting the bot and adding/removing reaction emojis, then verifying that roles are added/removed correspondingly

# Breaking Changes
None - if no existing saved message is found, the bot just continues as normal

### CODE REVIEW

File: commands/a_utility/pronouns.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 const { Command } = require('@sapphire/framework');
-const { Interaction, MessageEmbed, PermissionFlagsBits } = require('discord.js');
+const { Interaction, MessageEmbed, PermissionFlagsBits, Guild, Message, MessageManager } = require('discord.js');
 const firebaseUtil = require('../../db/firebase/firebaseUtil');
 
+var emojis = ['1️⃣', '2️⃣', '3️⃣', '4️⃣'];
```
Comment (byronwang93) [inline commands/a_utility/pronouns.js:5]: small comment but should we change this to `let`?

**Comment by byronwang93** [comment]
hmm not sure if you ran into this but when I run `/pronouns` I get this message

the steps I'm doing is running `/init-bot` and using this for the `roles.json` file:

[
    {
        "name": "hacker",
        "roleId": "1192271647634374748"
    },
    {
        "name": "sponsor",
        "roleId": "1192271647634374752"
    },
    {
        "name": "mentor",
        "roleId": "1192271647634374750"
    },
    {
        "name": "volunteer",
        "roleId": "1212287630893387784"
    },
    {
        "name": "photographer",
        "roleId": "1212287630893387783"
    }
]

Screen Shot 2024-07-29 at 2 45 42 PM

File: commands/a_utility/pronouns.js (hunk 1 above)
Comment (daniel-panhead) [reply commands/a_utility/pronouns.js]: Valid - changed

**Comment by daniel-panhead** [comment]
> hmm not sure if you ran into this but when I run `/pronouns` I get this message
> 
> the steps I'm doing is running `/init-bot` and using this for the `roles.json` file:
> 
> [ { "name": "hacker", "roleId": "1192271647634374748" }, { "name": "sponsor", "roleId": "1192271647634374752" }, { "name": "mentor", "roleId": "1192271647634374750" }, { "name": "volunteer", "roleId": "1212287630893387784" }, { "name": "photographer", "roleId": "1212287630893387783" } ]
> Screen Shot 2024-07-29 at 2 45 42 PM

That's really weird - this would only happen if the server didn't have the roles that perfectly matched all four pronoun strings `she/her`, `he/him`, `they/them`, and `other pronouns`. I didn't run into this issue

**Comment by byronwang93** [comment]
Ay ya that fixed it I needed to go into the server and manually create the new roles ^^

I also just committed some code in your PR to change up `pronouns.js` to create the roles if they don't already exist in the server, if you think this change is good feel free to merge your PR!

**Comment by daniel-panhead** [comment]
Nice!!
